dds_web/api/project.py

```diff
@@ -188,6 +188,107 @@ def post(self):
 
         return {"message": return_message}
 
+    @auth.login_required(role=["Unit Admin", "Unit Personnel"])
+    @logging_bind_request
+    @json_required
+    @handle_validation_errors
+    def patch(self):
+        """Partially update a the project status"""
+        # Get project ID, project and verify access
+        project_id = dds_web.utils.get_required_item(obj=flask.request.args, req="project")
+        project = dds_web.utils.collect_project(project_id=project_id)
+        dds_web.utils.verify_project_access(project=project)
+
+        # get atributes
+        json_input = flask.request.get_json(silent=True)  # Already checked by json_required
+
+        # false by default - operation must be confirmed by the user
+        confirmed_operation = json_input.get("confirmed", False)
+        if not isinstance(confirmed_operation, bool):
+            raise DDSArgumentError(message="`confirmed` is a boolean value: True or False.")
+        if not confirmed_operation:
+            warning_message = "Operation must be confirmed before proceding."
+            return {"warning": warning_message}
+
+        # Cannot change project status if project is busy
+        if project.busy:
+            raise ProjectBusyError(
+                message=(
+                    f"The deadline for the project '{project_id}' is already in the process of being changed. "
+                    "Please try again later. \n\nIf you know the project is not busy, contact support."
+                )
+            )
+
+        self.set_busy(project=project, busy=True)
+
+        try:
+            extend_deadline = json_input.get("extend_deadline", False)  # False by default
+
+            # some variable definition
+            curr_date = dds_web.utils.current_time()
+            send_email = False
+
+            # Update the deadline
+            if extend_deadline:
+                # deadline can only be extended from Available
+                if not project.current_status == "Available":
+                    raise DDSArgumentError(
+                        "you can only extend the deadline for a project that has the status Available."
+                    )
+
+                new_deadline_in = json_input.get("new_deadline_in")
+                current_deadline = (project.current_deadline - curr_date).days
+
+                if not new_deadline_in:
+                    raise DDSArgumentError(
+                        message="No new deadline provived, cannot perform operation."
+                    )
+                if current_deadline > 10:
+                    raise DDSArgumentError(
+                        message=f"There are still {current_deadline} days left, it is not possible to extend deadline yet."
+                    )
+                if new_deadline_in + current_deadline > 90:
+                    raise DDSArgumentError(
+                        message="The new deadline needs to be less than (or equal to) 90 days."
+                    )
```

Comment:
Maybe add a bit of information, something like this? 
```suggestion
                    raise DDSArgumentError(
                        message=f"You requested the deadline to be extended with {new_deadline_in} days (from {current_deadline}), giving a new total deadline of {new_deadline_in + current_deadline} days. The new deadline needs to be less than (or equal to) 90 days."
                    )
```